app/javascript/chat/ChatChannelSettings/MembershipManager/Membership.jsx

```diff
@@ -0,0 +1,128 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+
+import adminEmoji from '../../../../assets/images/emoji/apple-fire.png';
+
+const Membership = ({
+  membership,
+  currentMembership,
+  removeMembership,
+  handleUpdateMembershipRole,
+}) => {
+  const addAsModButton =
+    membership.role === 'member' ? (
+      <button
+        className="crayons-btn crayons-btn--s"
+        type="button"
+        onClick={handleUpdateMembershipRole}
+        data-membership-id={membership.membership_id}
+        data-role="mod"
+      >
+        Permote as Mod
+      </button>
+    ) : null;
+
+  const addAsMemberButton =
+    membership.role === 'mod' ? (
+      <button
+        className="crayons-btn crayons-btn--s"
+        type="button"
+        onClick={handleUpdateMembershipRole}
+        data-membership-id={membership.membership_id}
+        data-role="member"
+      >
+        Add as Member
```

Comment:
Since this means the member is a Mod and the action would __make them normal users__ (no longer Mods) I think this could be renamed to "Demote to Member" or "Remove Mod"